### PR TITLE
362 check if rasters use compression

### PR DESCRIPTION
**Pull request by elisalle**

### CODE REVIEW

File: threedi_modelchecker/checks/raster.py [hunk 1]
```diff
@@ -300,6 +300,20 @@ def description(self):
         return f"{self.column_name} has values {' and/or '.join(parts)} or is empty"
 
 
+class RasterCompressionUsedCheck(BaseRasterCheck):
+    """Checks whether compression was used for the raster"""
+
+    def is_valid(self, path: str, interface_cls: Type[RasterInterface]):
+        with interface_cls(path) as raster:
+            if not raster.is_valid_geotiff:
```
Comment (margrietpalm) [inline threedi_modelchecker/checks/raster.py:308]: Why is not being a valid geotiff sufficient for being compressed?

File: threedi_modelchecker/checks/raster.py [hunk 2]
```diff
@@ -300,6 +300,20 @@ def description(self):
...
+                return True
+            if raster.compression != "NONE":
+                return True
+
```
Comment (margrietpalm) [inline threedi_modelchecker/checks/raster.py:312]: Personally I'd add a `return False` to the end. This make it more clear, to me, that this functions tests a list of conditions and if none match than `is_valid` will return `False`.

File: threedi_modelchecker/interfaces/raster_interface_rasterio.py [hunk 3]
```diff
@@ -77,3 +77,7 @@ def min_max(self):
     @property
     def shape(self):
         return (self._dataset.height, self._dataset.width)
+
+    @property
+    def compression(self) -> Optional[str]:
```
Comment (margrietpalm) [inline threedi_modelchecker/interfaces/raster_interface_rasterio.py:82]: Return type doesn't match property in parent class

**Review by margrietpalm — APPROVED** [review]
Code looks fine. Assuming you created a test-case to confirm the exprected case is caught, this can be merged.

Please merge ASAP because this interferes with https://github.com/nens/threedi-modelchecker/pull/366

File: threedi_modelchecker/checks/raster.py (hunk 1 above)
Comment (elisalle) [reply threedi_modelchecker/checks/raster.py:308]: is_valid_geotiff returns False if the dataset is empty (such as when the file doesn't exist) or is not a valid geotiff. It's used on every raster check (such as RasterHasOneBandCheck) which reads the raster in any way, to prevent any other code in the check from executing if the dataset doesn't meet the require standards. If it doesn't exist, or isn't a valid geotiff, RasterExistsCheck and/or RasterIsValidCheck will be raised elsewhere.

File: threedi_modelchecker/interfaces/raster_interface_rasterio.py (hunk 3 above)
Comment (elisalle) [reply threedi_modelchecker/interfaces/raster_interface_rasterio.py]: thanks, fixed it

File: threedi_modelchecker/checks/raster.py (hunk 1 above)
Comment (margrietpalm) [reply threedi_modelchecker/checks/raster.py:308]: Ok, you may want to summarize this in a short comment

File: threedi_modelchecker/checks/raster.py (hunk 2 above)
Comment (margrietpalm) [reply threedi_modelchecker/checks/raster.py:312]: Great, given the explanation above that is actually much clear; the structure of the code then becomes:

* Ensure check should be performed on this path: if not return `False`
* Perform check

Comment (elisalle) [reply threedi_modelchecker/checks/raster.py:312]: I saw it slightly differently
Ensure check can be performed on this path; if not, return `True` so the check doesn't raise an exception
Perform check

File: threedi_modelchecker/checks/raster.py (hunk 1 above)
Comment (elisalle) [reply threedi_modelchecker/checks/raster.py:314]: I've put the comment at the top of the file because of it happening in all the checks.
